packages/typescript/internal/build_defs.bzl

```diff
@@ -831,4 +922,15 @@ def ts_library_macro(tsconfig = None, **kwargs):
         uses_plugin = kwargs.get("use_angular_plugin", False)
         supports_workers = not uses_plugin
 
-    ts_library(tsconfig = tsconfig, supports_workers = supports_workers, **kwargs)
+    is_windows = select({
+        "@bazel_tools//src/conditions:host_windows": True,
+        "//conditions:default": False,
+    })
+
+    # Unless this is set true, manually prevent JSON in the srcs attribute.
+    if not kwargs.get("allow_json_srcs", False):
```

Comment:
could you add some explanation why we need this, rather than always allow .json in srcs now?